From da50a12e0365bc1eee0ab59d45a1c275d927b739 Mon Sep 17 00:00:00 2001
From: Konstantin Tokarev <annulen@yandex.ru>
Date: Mon, 27 May 2024 23:23:11 +0300
Subject: [PATCH] Partial backport of r260554 (79fe19caf)

In particular, we need -DU_SHOW_CPLUSPLUS_API=0 to avoid compilation
errors in C++ parts of ICU headers which we are not using anyway.

Upstream-Status: Backport [https://github.com/qtwebkit/qtwebkit/commit/756e1c8f23dc2720471298281c421c0076d02df8 from https://github.com/qtwebkit/qtwebkit/issues/1105]

Change-Id: Ib45c74e3caad148fbd778d0c07330127f7dab5ec
---
 Source/WTF/wtf/Platform.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 7aefa1bab..a440aa218 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -1131,6 +1131,13 @@
 #define ENABLE_PLATFORM_FONT_LOOKUP 1
 #endif
 
+/* FIXME: This does not belong in Platform.h and should instead be included in another mechanism (compiler option, prefix header, config.h, etc) */
+/* ICU configuration. Some of these match ICU defaults on some platforms, but we would like them consistently set everywhere we build WebKit. */
+#define U_SHOW_CPLUSPLUS_API 0
+#ifdef __cplusplus
+#define UCHAR_TYPE char16_t
+#endif
+
 #if COMPILER(MSVC)
 #undef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS
